src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2096,25 +2141,49 @@ private MethodSymbol InferMethod(BoundCall node, MethodSymbol method, ImmutableA
             // PROTOTYPE(NullableReferenceTypes): Support field initializers in local functions.
         }
 
-        private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
+        /// <summary>
+        /// Returns the expression without the top-most conversion plus the conversion.
+        /// If the expression is not a conversion, returns the original expression plus
+        /// the Identity conversion. If `includeExplicitConversions` is true, implicit and
+        /// explicit conversions are considered. If `includeExplicitConversions` is false
+        /// only implicit conversions are considered and if the expression is an explicit
+        /// conversion, the expression is returned as is, with the Identity conversion.
+        /// (Currently, the only visit method that passes `includeExplicitConversions: true`
+        /// is VisitConversion. All other callers are handling implicit conversions only.)
+        /// </summary>
+        private static (BoundExpression Expression, Conversion Conversion) RemoveConversion(BoundExpression expr, bool includeExplicitConversions)
         {
-            // PROTOTYPE(NullableReferenceTypes): The loop is necessary to handle
-            // implicit conversions that have multiple parts (for instance, user-defined
-            // conversions). Should check for those cases explicitly.
+            ConversionGroup group = null;
             while (true)
             {
                 if (expr.Kind != BoundKind.Conversion)
                 {
                     break;
                 }
                 var conversion = (BoundConversion)expr;
-                if (conversion.ExplicitCastInCode || !conversion.Conversion.Exists)
+                if (group != conversion.ConversionGroupOpt && group != null)
                 {
+                    // E.g.: (C)(B)a
                     break;
                 }
+                group = conversion.ConversionGroupOpt;
+                Debug.Assert(group != null || !conversion.ExplicitCastInCode); // Explicit conversions should include a group.
+                if (!includeExplicitConversions && group?.IsExplicitConversion == true)
```

Comment:
>group?.IsExplicitConversion [](start = 51, length = 27)

Should this agree with ```conversion.ExplicitCastInCode```? Consider adding an assert and actually using ```conversion.ExplicitCastInCode``` here.

Reply:
`ExplicitCastInCode` may not be set on the first `BoundConversion` in the group.

---
In reply to: [190728151](https://github.com/dotnet/roslyn/pull/26356#discussion_r190728151) [](ancestors = 190728151)